### ISSUE TITLE
Edgemesh deployment optimization

### ISSUE THREAD

**Issue by JiaweiGithub** [kubeedge/kubeedge#4897]
**What would you like to be added/modified**:
1. Whether to consider enabling the "dynamicController" and "metaServer" configurations by default.
2. If the "metaServer" configuration cannot be enabled by default, can specify to enable it when keadm join.
3. Support to open the "metaServer" configuration when upgrading through the job.
4. When supporting keadm deployment, can choose to deploy Edgemesh together.

**Why is this needed**:
It is very inconvenient to change the variables of the cloud and edge when installing Edgemesh. Especially when there are a large number of edge nodes, it is very time-consuming to modify each one. Make some optimization suggestions.

**Comment by Poorunga** [comment]
Has the edgecore remote upgrade function of the keadm been developed? I think modifying edgecore configuration items also falls within this scope.

**Comment by ZhengXinwei-F** [comment]
The key point appears to be: 
point 1. Simplify the steps for enabling metaServer during edge node deployment.
point 2. Start the metaServer of each edge node that has joined the cluster in batches.

Here are my suggestions:
1. Because metaServer has performance requirements for edge nodes, it should not be enabled by default in keadm.
2. When keadm joins a node, it is a good idea to enable metaServer by configuration. Considering the large number of metaServer configuration entries, might we use the configuration file form?
3. In my opinion, the edgecore remote upgrade feature should include the function of point 2. Maybe we can discuss the scope and design of this feature.

**Comment by JiaweiGithub** [comment]
> 3\. Support to open the "metaServer" configuration when upgrading through the job.

The remote upgrade still cannot cover the second point. If want to install Edgemesh from the beginning, should not run another upgrade task. So I think the second point is the most needed at present.

**Comment by JiaweiGithub** [comment]
> 

The second point, I think other parameters can be filled in by default, only need to enable enable selection to be configurable. In addition, do we need to support security hardening by default?
https://edgemesh.netlify.app/guide/edge-kube-api.html#background

**Comment by JiaweiGithub** [comment]
> The key point appears to be: point 1. Simplify the steps for enabling metaServer during edge node deployment. point 2. Start the metaServer of each edge node that has joined the cluster in batches.
> 
> Here are my suggestions:
> 
> 1. Because metaServer has performance requirements for edge nodes, it should not be enabled by default in keadm.
> 2. When keadm joins a node, it is a good idea to enable metaServer by configuration. Considering the large number of metaServer configuration entries, might we use the configuration file form?
> 3. In my opinion, the edgecore remote upgrade feature should include the function of point 2. Maybe we can discuss the scope and design of this feature.

The second point, I think other parameters can be filled in by default, just open the enable option. In addition, do we need to support security hardening by default? 
https://edgemesh.netlify.app/guide/edge-kube-api.html#background

**Comment by ZhengXinwei-F** [comment]
> 
Yes. The configuration file and meshServer enable flags can both be opened. If no configuration file is given,  the default configuration will be used.
Ps: The security hardening feature might not good enough to be enabled by default. :)
